src/Framework/Profiler/ProfilerResult.cs

```diff
@@ -28,13 +28,11 @@ public ProfilerResult(IDictionary<EvaluationLocation, ProfiledLocation> profiled
         /// <inheritdoc />
         public override bool Equals(object obj)
         {
-            if (!(obj is ProfilerResult))
+            if (!(obj is ProfilerResult result))
```

Comment:
You can do the same trick as below:
return obj is ProfilerResult result &&
ProfiledLocations == result.ProfiledLocations...

Reply:
Yeah, I could, but I do not like the logic expression that I get in this case. It starts to have too complicated logic structure and for me it is more readable this way.